plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -133,14 +133,30 @@ void registerOptInStatus(boolean optedIn) {
    * If the user has never seen the opt-in dialog or set the opt-in preference beforehand,
    * this method can potentially present an opt-in dialog at the top workbench level. If you
    * are calling this method inside another modal dialog, consider using {@link #sendPing(
-   * String, String, String, Shell)} and pass the {@Shell} of currently open modal dialog.
+   * String, String, String, Shell)} and pass the {@Shell} of the currently open modal dialog.
    * (Otherwise, the opt-in dialog won't be able to get input until the workbench can get input.)
+   *
+   * Safe to call from non-UI contexts.
    */
   public void sendPing(String eventName, String metadataKey, String metadataValue) {
     sendPing(eventName, metadataKey, metadataValue, null);
   }
 
-  public void sendPing(String eventName, String metadataKey, String metadataValue,
+  public void sendPing(final String eventName,
+      final String metadataKey, final String metadataValue, final Shell parentShell) {
+    if (Display.getCurrent() != null) {
```

Comment:
Actually, what I really wanted to know was that if I am in the UI thread or not. (I thought `Display.getCurrent() != null()` served the purpose.) It's about whether I should wrap the main body inside `Display.asyncExec()` or not, to avoid the SWT thread access issue.

I could certainly use `PlatformUI.getWorkbench().getDisplay()`. (Initially, I got the idea of `Display#getDefault()` from [here](https://github.com/GoogleCloudPlatform/gcloud-eclipse-tools/blob/master/plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/Activator.java#L68)).